File: algorithms/2_medium/S/02785_sort_vowels_in_a_string/normal_sort.cpp
class Solution {
    inline bool is_vowel(char ch) {
        return (ch == 'a' || ch == 'e' || ch =='i' || ch =='o' || ch == 'u'
            || ch == 'A' || ch == 'E' || ch =='I' || ch =='O' || ch == 'U');
    }
public:
    string sortVowels(string s) {
        vector<char> vec_stack{};

        for (auto &ch: s) {
            if (is_vowel(ch)) {
                vec_stack.push_back(ch);
            }
        }

        sort(vec_stack.rbegin(), vec_stack.rend());

        for (char &ch: s) {
            if (is_vowel(ch)) {
                ch = vec_stack.back();
                vec_stack.pop_back();
            }
        }

        return s;
    }
};

File: algorithms/2_medium/S/02785_sort_vowels_in_a_string/normal_sort.rs
impl Solution {

    #[inline]
    fn is_vowel(ch: &u8) -> bool {
        matches!(*ch, b'a' | b'e' | b'i' | b'o' | b'u' | b'A' | b'E' | b'I' | b'O' | b'U')
    }

    pub fn sort_vowels(mut s: String) -> String {
        let mut s_bytes = s.into_bytes();

        //inserting all vowel chars of current string into vector
        let mut vowels: Vec<u8> = s_bytes.iter().filter(|&ch| Self::is_vowel(ch)).cloned().collect();
        vowels.sort_unstable_by(|a, b| b.cmp(a)); // then sorting that vector

        for byte in s_bytes.iter_mut() {
            if Self::is_vowel(byte) {
                *byte = vowels.pop().unwrap();
            }
        }

        String::from_utf8_lossy(&s_bytes).into_owned()
    }
}
